src/main/java/Model/Game.java

```diff
@@ -1,269 +1,383 @@
      */
-    private boolean checkForEnPassant(Position previousPosition, Position newPosition, Piece piece) {
-        if (piece.getPieceType() == PieceType.Pawn) {
-            if(enPassantAvailablePieces.contains(piece)) {
-                if(Objects.equals(piece.getEnPassantAvailableToTakeFile(), newPosition.getFile())) {
-                    this.removePiece(previousPosition);
-                    this.removePiece(new Position(newPosition.getFile(), previousPosition.getRank()));
-                    this.addPiece(newPosition, piece);
-                    nextTurn();
-                    disableEnPassant();
-                    updateView(newPosition, previousPosition);
-
-                    return true;
-                }
-            }
-        }
-        return false;
+    private ArrayList<Piece> getColouredPieces(boolean isWhite){
+        return isWhite?whitePieces:blackPieces;
     }
 
+
+
     /**
      * Remove the availability to en Passant from all the necessary pieces
      */
     private void disableEnPassant() {
         for (Piece piece:enPassantAvailablePieces) {
             piece.setEnPassantAvailableToTakeFile(null);
         }
-        this.enPassantPosition = null;
+        this.enPassantPositionForFen = null;
         enPassantAvailablePieces.clear();
     }
 
+
+    /**
+     * Update any pawn that can en Passant next turn and then add those pieces that
+     * are able to en passant on the next turn to the enPassantAvailablePieces list
+     * 
+     * @param newPosition new position of the pawn
+     * @param isWhite if the pawn is white or black
+     */
+    
+    private void enableEnPassant(Position newPosition, boolean isWhite) {
+        int tempFile;
+        for (int iFile = -1; iFile < 2; iFile+=2) {
+            tempFile = newPosition.getFile() + iFile;
+            if (tempFile >= 0 && tempFile < 8){
+                Piece piece = board.getPiece(new Position(tempFile,newPosition.getRank()));
+                if(nonNull(piece) && piece.getPieceType() == PieceType.Pawn && piece.getIsWhite() != isWhite){
+                    piece.setEnPassantAvailableToTakeFile(newPosition.getFile());
+                    enPassantAvailablePieces.add(piece);
+                }
+            }
+        }
+    }
+
+
     /**
+     * WIP!
      * Check to see if a pawn has made it to the other side of the board. If so call the promotion function
      * 
-     * @param newRank
-     * @param piece
+     * @param newPosition of the piece
+     * @param piece that just moved
      */
-    private void checkForPromotion(int newRank, Piece piece) {
+    private void checkForPromotion(Position newPosition, Piece piece) {
         if(piece.getPieceType() != PieceType.Pawn) {
             return;
         }
         int rank = piece.getIsWhite() ? 0 : 7;
-        if(newRank != rank){
+        if(newPosition.getRank() != rank){
             return;
         }
         piece.pawnPromotion(PieceType.Queen);
     }
```

Comment:
hmm dont love calling this `checkForPromotion()`, because you aren't just checking. If you find its promotable, you then actually perform the promotion. I'd probably make this return a boolean, and use it like:
```java
boolean canPromote = checkForPromotion(position, piece);
if (canPromote) {
    piece.pawnPromote()
}
```
Even better would be to make `canPromote()` a method on the `Piece`:
```java
if (piece.canPromote()) {
    piece.promote();
}
```

Reply:
Will change it when I work on promotion. Had it there ages ago